homeassistant/components/sensor/rejseplanen.py

```diff
@@ -0,0 +1,221 @@
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['rjpl==0.3.2']
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_STOP_ID = 'Stop ID'
+ATTR_STOP_NAME = 'Stop'
+ATTR_ROUTE = 'Route'
+ATTR_TYPE = 'Type'
+ATTR_DIRECTION = "Direction"
+ATTR_DUE_IN = 'Due in'
+ATTR_DUE_AT = 'Due at'
+ATTR_NEXT_UP = 'Later departure'
+
+CONF_ATTRIBUTION = "Data provided by rejseplanen.dk"
+CONF_STOP_ID = 'stop_id'
+CONF_ROUTE = 'route'
+CONF_DIRECTION = 'direction'
+CONF_DEPARTURE_TYPE = 'departure_type'
+
+DEFAULT_NAME = 'Next departure'
+ICON = 'mdi:bus'
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STOP_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_ROUTE, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_DIRECTION, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_DEPARTURE_TYPE, default=[]):
+        vol.All(cv.ensure_list, [vol.In(list(['BUS', 'EXB', 'M',
+                                              'S', 'REG']))])
+})
+
+
+def due_in_minutes(timestamp):
+    """Get the time in minutes from a timestamp.
+
+    The timestamp should be in the format day.month.year hour:minute
+    """
+    diff = datetime.strptime(
+        timestamp, "%d.%m.%y %H:%M") - dt_util.now().replace(tzinfo=None)
+
+    return int(diff.total_seconds() // 60)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Rejseplanen transport sensor."""
+    name = config.get(CONF_NAME)
+    stop_id = config.get(CONF_STOP_ID)
+    route = config.get(CONF_ROUTE)
+    direction = config.get(CONF_DIRECTION)
+    departure_type = config.get(CONF_DEPARTURE_TYPE)
+
+    data = PublicTransportData(stop_id, route, direction, departure_type)
+    add_devices([RejseplanenTransportSensor(data,
+                                            stop_id,
+                                            route,
+                                            direction,
+                                            name)],
+                True)
+
+
+class RejseplanenTransportSensor(Entity):
+    """Implementation of Rejseplanen transport sensor."""
+
+    def __init__(self, data, stop_id, route, direction, name):
+        """Initialize the sensor."""
+        self.data = data
+        self._name = name
+        self._stop_id = stop_id
+        self._route = route
+        self._direction = direction
+        self._times = self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._times is not None:
+            next_up = "None"
```

Comment:
Usually we don't include items in the state attributes, evaluated per update, where the value is not known.

Reply:
So the  `ATTR_NEXT_UP` key should not be set at all in that case, or should it be actual None (without quotes).

Reply:
It should not be set at all.